From 78be1aaa2dee57d7f7457c559984142a6321166c Mon Sep 17 00:00:00 2001
From: Ran Shalit <rshalit@habana.ai>
Date: Tue, 01 Dec 2020 16:14:11 +0200
Subject: [PATCH] [SW-25369] gaudi: add key index into boot info

Add key index from fit into boot info, so that this
information shall be available in u-boot.

Change-Id: I4f456d343eb7c3d2595dff2e0ffbe53e5ae15a9f
Signed-off-by: Ran Shalit <rshalit@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 8d2cec8..95fea41 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1504,3 +1504,8 @@
 
 	preboot_secure_hbw_tpc3_seclvl_open();
 }
+
+void preboot_security_sp_sram_write_key_idx(u32 *key_idx)
+{
+	boot_info_set(HL_BOOT_INFO_KEY_IDX, key_idx, sizeof(u32));
+}
diff --git a/preboot/fit.c b/preboot/fit.c
index 632c59a..8609515 100644
--- a/preboot/fit.c
+++ b/preboot/fit.c
@@ -331,10 +331,14 @@
 	return fdt_subnode_offset(fit, images, FIT_PUB_KEY_EXT_PROP);
 }
 
+__weak void preboot_security_sp_sram_write_key_idx(u32 *key_idx) {}
+
+static u32 preboot_key_idx;
+
 static int fit_get_pub_key(const void *fit, int noffset, int sig_sub_node,
 			   const void **pub_key_fit)
 {
-	int node, key_idx, key_len, pub_key_node;
+	int node, key_len, pub_key_node;
 	const void *key_ext_data = NULL;
 	const void *key_data;
 	const char *key_name;
@@ -352,8 +356,8 @@
 		return 1;
 	}
 
-	key_idx = fit_get_int(fit, 0, FIT_SEC_KEY_IDX, -1);
-	if (key_idx < 0) {
+	preboot_key_idx = fit_get_int(fit, 0, FIT_SEC_KEY_IDX, -1);
+	if (preboot_key_idx < 0) {
 		preboot_puts("Wrong FIT format: missing key idx\n");
 		return 1;
 	}
@@ -396,7 +400,7 @@
 	}
 
 key_validation:
-	if (!preboot_security_key_is_valid(key_idx, key_data, key_len,
+	if (!preboot_security_key_is_valid(preboot_key_idx, key_data, key_len,
 				key_ext_data, key_ext_len)) {
 		preboot_puts("Key validation failed\n");
 		return 1;
@@ -605,6 +609,8 @@
 		fit_copy_image(load, buf, len, images[i].arch);
 	}
 
+	preboot_security_sp_sram_write_key_idx(&preboot_key_idx);
+
 	return FIT_LOAD_SUCCESS;
 }
 
